server/src/main/java/io/crate/collections/accountable/AccountableList.java

```diff
@@ -0,0 +1,189 @@
+            Arrays.sort((T[]) root.elementData, offset, offset + size, c);
+        }
+
+        @SuppressWarnings("unchecked")
+        public T set(int index, T element) {
+            Objects.checkIndex(index, size);
+            T oldValue = (T) root.elementData[offset + index];
+            root.elementData[offset + index] = element;
+            return oldValue;
+        }
+
+        @SuppressWarnings("unchecked")
+        public T get(int index) {
+            Objects.checkIndex(index, size);
+            return (T) root.elementData[offset + index];
+        }
+
+        public int size() {
+            return size;
+        }
+
+        public boolean add(T element) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    /**
+     * Simplified version of {@link ArrayList#grow(int)}} operation but with minGrowth = 1.
+     * We don't support all-at-once version of "addAll" or "ensureCapacity", so minGrowth is fixed and equals to 1.
+     *
+     */
+    private Object[] grow(int minCapacity) {
+        int oldCapacity = elementData.length;
+        if (oldCapacity > 0 || elementData != DEFAULTCAPACITY_EMPTY_ELEMENTDATA) {
+            int newCapacity = newLength(oldCapacity, oldCapacity >> 1);
+            // Same as RamUsageEstimator.shallowSizeOf(array) but without NUM_BYTES_ARRAY_HEADER as we are accounting only for expansion.
+            allocateBytes.accept(RamUsageEstimator.alignObjectSize((long) NUM_BYTES_OBJECT_REF * (newCapacity - oldCapacity)));
+            elementData = Arrays.copyOf(elementData, newCapacity);
+        } else {
+            elementData = new Object[Math.max(DEFAULT_CAPACITY, minCapacity)];
+            allocateBytes.accept(RamUsageEstimator.shallowSizeOf(elementData));
+        }
+        return elementData;
+    }
+
+    /**
+     * Copy of ArraysSupport.newLength with following preconditions:
+     * minGrowth = 1. prefGrowth >= minGrowth.
+     */
+    private static int newLength(int oldLength, int prefGrowth) {
+        int prefLength = oldLength + prefGrowth; // might overflow
+        if (prefLength <= SOFT_MAX_ARRAY_LENGTH) {
+            return prefLength;
+        } else {
+            return hugeLength(oldLength);
+        }
+    }
+
+    /**
+     * Copy of to ArraysSupport.hugeLength with fixed minGrowth = 1.
+     * Despite on accounting, we still throw OOM here.
+     * This is done to satisfy general VM contract that arrays with size > MAX_INT cannot be allocated even if there is enough memory.
+     */
+    private static int hugeLength(int oldLength) {
+        int minLength = oldLength + 1;
+        if (minLength < 0) { // overflow
+            throw new OutOfMemoryError(
+                "Required array length " + oldLength + " + " + 1 + " is too large");
+        } else if (minLength <= SOFT_MAX_ARRAY_LENGTH) {
+            return SOFT_MAX_ARRAY_LENGTH;
+        } else {
+            return minLength;
+        }
+    }
```

Comment:
Why do we need to implement these? Can't we just call `ArraysSupport.newLength()` with the `minGrowth` set to `1` instead?

Reply:
I copied it because we can't import it:

> Package 'jdk.internal.util' is declared in module 'java.base', which does not export it to the unnamed module

Would have to add `--add-exports...` to module compiler options - I thought its a bit overkill given that methods are small and I anyway customized them.

Reply:
Ah ok, thanks.